cmd/check_licenses.go

```diff
@@ -0,0 +1,150 @@
+// Copyright © 2016 Prometheus Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// validSourceExtensions is a slice of strings
+	// representing the file suffix of source code files to inspect.
+	validSourceExtensions []string
+
+	// headerLength is the number of lines to read in.
+	headerLength int
+
+	// defaultSourceExtensions is a slice of strings
+	// representing the file suffix of source code files to inspect
+	// defaulted to only include go source files.
+	defaultSourceExtensions = []string{".go"}
+
+	// validHeaderStrings is a slice of strings that must exist in a header.
+	validHeaderStrings = []string{"copyright", "generated"}
+
+	// checkLicensesCmd represents the check_licenses command.
+	checkLicensesCmd = &cobra.Command{
+		Use:   "check_licenses [<location>]",
```

Comment:
We should be consistent with Prometheus/promtool where we use spaces for subcommands: `promu check licenses ...`

Reply:
so in that case we'd have a command that attaches to the root Promu command and this license check command is attached to the check subcommand?

Reply:
ok, just wanted to confirm

Reply:
Yep that's it.